types/ember/test/debug.ts

```diff
@@ -4,10 +4,16 @@ const {
     runInDebug,
     warn,
     debug,
-    assert,
     Debug: { registerDeprecationHandler, registerWarnHandler }
 } = Ember;
 
+// Workaround for https://github.com/microsoft/TypeScript/issues/36931.This is
+// still safe to use in the tests, because if we cast as something that (a)
+// isn't on the namespace or (b) doesn't match the definition which *is* on the
+// namespace, this will not type-check.
+const assertDescOnly: (desc: string) => never = Ember.assert;
+const assertFull: (desc: string, test: unknown) => asserts test = Ember.assert;
```

Comment:
I think this will also work (since the rule is really _just_ "there must be a type annotation"):

```ts
const assert: typeof Ember.assert = Ember.assert;
```

https://www.typescriptlang.org/play/index.html?ssl=6&ssc=1&pln=7&pc=1#code/CYUwxgNghgTiAEA7KBbEBnADlMCCiKARiDPAN4BQ818AZgK6JgAuAlgPaLxTronMAKUOjAAueOmYxWiAOYBKcYhAA3EgG4qNBkzaduvfkIxiJUmbIA08ZhmbjGAa0TsA7okUG+MZuht3NAF8KEIowTkkvfnFmAE9MEHZaeAJiGAA6Hm9meABeFKISTMMfTVDQSFgECBAcgA9xSWk5eAAfJHoICE0sowByOvhWPyl6ZgALWL7rOvkyuvTmdgBVTASYAGEeEAE5+AB6fe4u+Fl2dmB1eFcEXp8QYHhB4ZsYMcnQu8E+wnOUAEI+nMwhF2DV0hB2LIBH0Ji8Xow4DhxlBCDV4OFQNxEI90ON2J1HsR4ONWLJxhAyeNbI8eBJ6GBxkDNEA

Reply:
Ah, very good. (I thought I had tried this, but I think I actually tried using a cast `as typeof Ember.assert` instead.)